### DISCUSSION TITLE
Intercepting internal server error with useFetch

### DISCUSSION THREAD

**Discussion by Mensix** [nuxt/nuxt#18235] [Questions]
Not sure if it should go here or my thinking is good, let's try anyway. I'm looking for a way to intercept 500 server error, called for example when response called from useFetch returns 404 (so the page can't be rendered with given data in template). Currently, the nuxtish 500 page is returned, but I would like to customize it, like provide information that the data was not found in a normal way. Any thoughts?

Note: it seems like it was also the case in Nuxt2 [#](https://stackoverflow.com/questions/58325686/how-to-handle-nuxt-ssr-errors-and-show-custom-404-500-pages).

Minimal reproduction: https://stackblitz.com/edit/nuxt-starter-wt1szj (go to /todo/1, then /todo/1000).

**Comment by Mensix** [comment]
Is this it? :)

![image](https://user-images.githubusercontent.com/44240517/137584578-1a005995-22e2-43d5-817d-4ce233ebff91.png)

**Reply by pi0** [reply]
For intercepting, you can create highier order composable on top of `useFetch`:

```vue

```vue
<script setup>
const callAPI= (req, opts) =>
 useFetch(req, { ...opts, headers: { test: '123' } })
     .catch((err) => ({ error: err }));

const { data, refresh } = await callAPI('/api/test');
</script>
```

**Comment by AndersNielsen85** [comment]
Would be a great feature to have in useFetch indeed. I struggled to do the same a few weeks ago.
Ended up using useAsyncData(for now) instead, like this:
```
const { data } = await useAsyncData('myRandomKey', () => $fetch('https://httpstat.us/500').catch((e) => {
  console.log(e)
}));
```

**Reply by pi0** [reply]
Do you mean https://v3.nuxtjs.org/docs/usage/data-fetching#usefetch? :D

**Reply by AndersNielsen85** [reply]
@pi0 To me it seems like .catch does nothing when using useFetch. 

Produces no console log
```
const { data } = await useFetch('https://httpstat.us/404').catch((e) => {
  console.log('error from fetch', e)
});
```

Produces a console log
```
const { data } = await useAsyncData('asyncData', () => $fetch('https://httpstat.us/404').catch((e) => {
  console.log('error from async', e)
}));
```

The docs mention that the API response types are inferred with useFetch, so I presume it is because errors are already dealt with inside the useFetch basically disabling the .catch?

**Reply by danielroe** [reply]
@AndersNielsen85 `useFetch` and `useAsyncData` return `error`, in addition to `data`, `refresh`, etc. You can inspect this object to handle errors.

**Reply by AndersNielsen85** [reply]
@danielroe Thanks for this
```
const { data, error } = await useFetch('https://httpstat.us/404');
if (error.value) {
  console.log('fetchERR', error.value)
}
```
Indeed this works :)

Maybe some error-handling (and for the etc.) could be added to documentation at some point?

**Reply by pi0** [reply]
https://github.com/nuxt/framework/issues/1580
